twpca/model.py

```diff
@@ -54,152 +90,249 @@ def __init__(self, n_components, shared_length=None,
             'warp': warp_regularizer,
         }
 
-        # store tensorflow variables, parameters and session
-        self._raw_params = {} # tensorflow variables
-        self._params = {}     # model parameters (if nonneg=True, softplus transform applied to raw_params)
-        self._sess = None
+        # create all tensorflow variables
+        self._train_vars = {} # tensorflow variables that are optimized
+        # factor matrix parameters
+        self._train_vars['time'] = tf.Variable(tf.zeros((self.n_timepoints, self.n_components)))
+        self._train_vars['neuron'] = tf.Variable(tf.zeros((self.n_neurons, self.n_components)))
+        if self.fit_trial_factors:
+            self._train_vars['trial'] = tf.Variable(tf.zeros((self.n_neurons, self.n_components)))
+        # warp parameters
+        self.tau = tf.Variable(tf.zeros((self.n_trials, self.shared_length)))
+        self.tau_shift = tf.Variable(tf.zeros(self.n_trials))
+        self.tau_scale = tf.Variable(tf.zeros(self.n_trials))
+        # learning rate for optimizers
+        self._lr = tf.placeholder(tf.float32, shape=[])
```

Comment:
add doc here (or maybe at the top of the file?) that class variables prefixed with _ are TensorFlow Tensors